test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -41,6 +44,11 @@
 (deftest ^:integration old-master-route-config
   (testing "The old map-style route configuration map still works."
     (bootstrap/with-puppetserver-running-with-mock-jrubies
+     "MOCK.TODO: discuss - this is *probably* ok because all we really care about
+     is that the webserver is accepting connections and handing them to the correct
+     ring handler.  Would be better if we could implement this to hit a clojure
+     endpoint instead of a Ruby one but I don't remember if there are any pure-clojure
+     endpoints under /puppet."
```

Comment:
Yeah, I think I could do that.  At a lower-level, I already have an optional argument to the `mock-jruby-pool-manager-service` function which takes in a whole `mock-jruby-puppet-fn` for creating the JRubyPuppet instance.  For a little more flexibility, I could add an optional parameter to the `create-mock-jruby-puppet` function which is the `request-handler` function that you want it to call back to.  Then maybe I could plumb an optional `mock-jruby-puppet-fn` from `with-puppetserver-running-with-mock-jrubies` on through to the `mock-jruby-pool-manager-service`.  The advantage of that vs. what you're suggesting is that individual tests could also mock the other JRubyPuppet functions as well, if needed.  WDYT?